app/src/main/java/com/jvoye/tasky/auth/presentation/register/RegisterScreen.kt

```diff
@@ -51,20 +51,24 @@ import org.koin.androidx.compose.koinViewModel
 
 @Composable
 fun RegisterScreenRoot(
+    onLogInClick: () -> Unit,
+    onSuccessfulRegistration: () -> Unit,
     viewModel: RegisterViewModel = koinViewModel()
 ) {
     val state by viewModel.state.collectAsStateWithLifecycle()
 
     RegisterScreen(
         state = state,
-        onAction = viewModel::onAction
+        onAction = viewModel::onAction,
```

Comment:
Suggestion 🙂 Keep a single onAction and intercept only the login action with a when at the call site. Simple, clear, and keeps navigation out of the ViewModel.
```kotlin
RegisterScreen(
    state = state,
    onAction = { action ->
        when (action) {
            RegisterAction.OnLoginClick -> onLogInClick()
            else -> viewModel.onAction(action)
        }
    }
)
```